examples/closure-app/silabs/src/ClosureUI.cpp

```diff
@@ -0,0 +1,240 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "AppConfig.h"
+#include "ClosureUI.h"
+#include "demo-ui-bitmaps.h"
+#include "dmd.h"
+#include "glib.h"
+#include "lcd.h"
+
+// Only needed for wifi NCP devices
+#if SL_WIFI && !defined(SLI_SI91X_MCU_INTERFACE)
+#include <platform/silabs/wifi/ncp/spi_multiplex.h>
+#endif // SL_WIFI && !defined(SLI_SI91X_MCU_INTERFACE)
+
+// Layout positions
+#define STATUS_ICON_LINE 0
+#define SILABS_ICON_POSITION_X 0
+#define BLE_ICON_POSITION_X 72
+#define NETWORK_ICON_POSITION_X 90
+#define MATTER_ICON_POSITION_X 108
+
+// Icon sizes
+#define SILABS_LOGO_WIDTH 47
+#define SILABS_LOGO_HEIGHT 18
+#define BLUETOOTH_ICON_SIZE 18
+
+// State display positions
+#define STATE_DISPLAY_LINE 3
+#define POSITION_DISPLAY_LINE 5
+#define LATCH_DISPLAY_LINE 6
+#define SECURE_DISPLAY_LINE 7
+#define SPEED_DISPLAY_LINE 8
+#define FOOTER_DISPLAY_LINE 11
+
+// Bitmap definitions
+static const uint8_t silabsLogo[]       = { SILABS_LOGO_SMALL };
+static const uint8_t matterLogoBitmap[] = { MATTER_LOGO_BITMAP };
+static const uint8_t wifiLogo[]         = { WIFI_BITMAP };
+static const uint8_t threadLogo[]       = { THREAD_BITMAP };
+static const uint8_t bleLogo[]          = { BLUETOOTH_ICON_SMALL };
+
+// Static variables to hold the current closure state
+ClosureUI::MainState ClosureUI::sMainState = ClosureUI::STATE_UNKNOWN;
+char ClosureUI::sPositionText[24]          = "Position: Unknown";
```

Comment:
Why those string duplicated ? if they need to be reused we should put them either in static variable or in defines instead of having them hardcoded in two different files.